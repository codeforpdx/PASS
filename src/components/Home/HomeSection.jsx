// React Imports
import React from 'react';
// Material UI Imports
import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';
import Button from '@mui/material/Button';

/**
 * Represents a home section component
 *
 * @memberof Home
 * @name HomeSecton
 * @param {object} Props - the component props
 * @param {string} Props.componentImageSrc - image src
 * @param {string} Props.componentImageAlt - image alt
 * @param {string} Props.title - section title
 * @param {string} Props.description - section description
 * @param {string} Props.button - section button
 * @param {string} Props.href - section button href
 * @param {string} Props.label - section button aria-label
 * @param {boolean} Props.isReallySmallScreen - screen size
 * @returns {React.JSX.Element} - the home section component
 */
const HomeSection = ({
  componentImageSrc,
  componentImageAlt,
  title,
  description,
  button,
  href,
  label,
  isReallySmallScreen
}) => (
  <>
    <Box
      component="img"
      src={componentImageSrc}
      alt={componentImageAlt}
      sx={{
        width: isReallySmallScreen ? '80%' : '300px'
      }}
    />
    <Typography
      variant="h2"
      sx={{
        color: 'primary.main',
        textAlign: 'center',
        marginBottom: '24px',
        fontSize: '28px'
      }}
    >
      <strong>{title}</strong>
    </Typography>
    <Typography
      sx={{
        color: 'primary.dark',
        width: isReallySmallScreen ? '100%' : '85%',
        marginBottom: '24px',
        textAlign: 'center'
      }}
    >
      {description}
    </Typography>
    {button && (
      <Button
        variant="contained"
        href={href}
        aria-label={label}
        sx={{
          my: '1rem',
          backgroundColor: 'primary.light',
          width: isReallySmallScreen ? 1 : 1 / 4,
          borderRadius: '25px'
        }}
      >
        {button}
      </Button>
    )}
  </>
);

export default HomeSection;
