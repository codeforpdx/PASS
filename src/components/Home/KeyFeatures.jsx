// React Imports
import React from 'react';
// Material UI Imports
import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';
/**
 * @typedef {object} KeyFeaturesParams
 * @property {string} icon - icon
 * @property {string} title - key feature title
 * @property {string} description - key feature description
 * @property {boolean} isReallySmallScreen - screen size
 */

/**
 * Represents key features section component
 *
 * @memberof Home
 * @name KeyFeatures
 * @param {KeyFeaturesParams} props - the props
 * @returns {React.JSX.Element} the KeyFeatures section component
 */

const KeyFeatures = ({ icon, title, description, isReallySmallScreen }) => (
  <>
    <Box
      sx={{
        display: 'flex',
        alignItems: 'center',
        marginBottom: '14px',
        color: 'primary.main',
        width: isReallySmallScreen ? 1 : 'full'
      }}
    >
      {icon}
      <Typography
        variant={isReallySmallScreen ? 'h5' : 'h4'}
        sx={{
          marginLeft: '8px',
          textAlign: isReallySmallScreen ? 'start' : 'center',
          width: '100%'
        }}
      >
        <strong>{title}</strong>
      </Typography>
    </Box>
    <Typography
      variant="body1"
      sx={{
        width: isReallySmallScreen ? 1 : 2 / 3,
        textAlign: isReallySmallScreen ? 'start' : 'center',
        color: 'primary.dark',
        marginBottom: '50px'
      }}
    >
      {description}
    </Typography>
  </>
);

export default KeyFeatures;
