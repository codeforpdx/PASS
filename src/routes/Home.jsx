// React Imports
import React from 'react';
// Material UI Imports
import Box from '@mui/material/Box';
import Container from '@mui/material/Container';
import Paper from '@mui/material/Paper';

/**
 * Home Component - Component that generates the home page of the PASS website/app
 *
 * @memberof Login
 * @name Login
 */

const Home = () => (

  /* eslint-disable jsx-a11y/label-has-associated-control */
    <Container component="main" maxWidth="s">
      <Box
        sx={{
          marginTop: 18,
          display: 'flex',
          flexDirection: 'column',
          alignItems: 'center',
          textAlign: 'center',
          padding: '20px'
        }}
      >
        <Paper elevation={2} sx={{ display: 'inline-block', mx: '2px', padding: '20px' }}>
          <section id="home">
            <h1>HOME PAGE</h1>
          </section>
        </Paper>
      </Box>
    </Container>
  /* eslint-enable jsx-a11y/label-has-associated-control */
)

export default Home;
