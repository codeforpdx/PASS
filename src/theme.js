import { createTheme } from '@mui/material/styles';

const theme = createTheme({
  typography: {
    fontFamily: 'Roboto, sans-serif',
    dd: { fontWeight: 'bold' },
    dl: {
      display: 'grid',
      gridTemplateColumns: 'max-content auto'
    }
  },
  palette: {
    primary: {
      light: '#039686',
      main: '#017969',
      dark: '#004d3e',
      slight: '#8fc2bb',
      contrastText: '#fff'
    },
    secondary: {
      light: '#b32126',
      main: '#961020',
      dark: '#790111',
      contrastText: '#fff'
    },
    tertiary: {
      light: '#dbc584',
      main: '#DEBC59',
      dark: '#dbb032',
      contrastText: '#fff'
    },
    status: {
      danger: '#e53e3e'
    },
    neutral: {
      main: '#64748B',
      contrastText: '#fff'
    }
  }
});

export default theme;
