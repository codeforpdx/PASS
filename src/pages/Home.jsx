// React Imports
import React, { useRef } from 'react';
// Material UI Imports
import ArrowCircleDownOutlinedIcon from '@mui/icons-material/ArrowCircleDownOutlined';
import Box from '@mui/material/Box';
import Container from '@mui/material/Container';
import Diversity1Icon from '@mui/icons-material/Diversity1';
import IconButton from '@mui/material/IconButton';
import SecurityIcon from '@mui/icons-material/Security';
import Stack from '@mui/material/Stack';
import SupportIcon from '@mui/icons-material/Support';
import Typography from '@mui/material/Typography';
import useMediaQuery from '@mui/material/useMediaQuery';
import { useTheme } from '@mui/material/styles';
import { visuallyHidden } from '@mui/utils';
// Components Import
import { HomeSection, KeyFeatures } from '@components/Home';

/**
 * Home - First Page you encounter in PASS before login.
 * Should not display if you are already logged in.
 *
 * @memberof Pages
 * @name Home
 * @returns {React.ReactElement} The home page
 */
const Home = () => {
  const theme = useTheme();
  const isReallySmallScreen = useMediaQuery(theme.breakpoints.down('sm'));
  const iconSize = isReallySmallScreen ? 'medium' : 'large';
  const aboutRef = useRef(null);

  const handleClick = () => aboutRef.current.scrollIntoView({ behavior: 'smooth' });

  // Logo section elements
  const heading = (
    <>
      PASS
      {/* This is added for better screen reader experience by adding a pause
          between the acronym and the expanded acronym. */}
      <Box component="span" sx={visuallyHidden}>
        :
      </Box>
    </>
  );
  const subheading = 'Personal Access System for Services';

  const logoSection = isReallySmallScreen ? (
    <Typography component="h1" color="primary">
      <Stack
        component="span"
        justifyContent="center"
        alignItems="center"
        spacing={2}
        mb={2}
        data-testid="testStack"
      >
        <Typography variant="h1" component="span" fontWeight="500" fontSize="72px">
          {heading}
        </Typography>
        <Box component="img" src="/assets/PASSLogolightmode.png" alt="" width="50%" />
        <Typography variant="h4" component="span" fontWeight="600" mb={8}>
          {subheading}
        </Typography>
      </Stack>
    </Typography>
  ) : (
    <Typography component="h1" color="primary">
      <Stack
        justifyContent="center"
        alignItems="center"
        spacing={4}
        mb={18}
        data-testid="testStack"
      >
        <Stack
          component="span"
          direction="row"
          justifyContent="center"
          alignItems="center"
          spacing={4}
        >
          <Box component="img" src="/assets/PASSLogolightmode.png" alt="" width="150px" />
          <Typography variant="h1" component="span" fontWeight="500" fontSize="144px">
            {heading}
          </Typography>
        </Stack>
        <Typography variant="h3" component="span" fontWeight="600">
          {subheading}
        </Typography>
      </Stack>
    </Typography>
  );

  return (
    <Container sx={{ width: '100vw' }}>
      <Box
        sx={{
          margin: '18px 0',
          display: 'flex',
          flexDirection: 'column',
          alignItems: 'center',
          textAlign: 'center',
          padding: '20px'
        }}
      >
        <section id="home" data-testid="testHomeSection">
          <Stack alignItems="center" justifyContent="center">
            <Box
              height="calc(100vh - 64px)"
              display="flex"
              flexDirection="column"
              justifyContent="center"
              alignItems="center"
            >
              {logoSection}
              <Typography color="secondary.main" sx={{ fontSize: '24px' }}>
                Learn More
              </Typography>
              <IconButton aria-label="Scroll down" color="secondary" onClick={handleClick}>
                <ArrowCircleDownOutlinedIcon sx={{ fontSize: '32px' }} />
              </IconButton>
            </Box>
            <div ref={aboutRef}>
              <HomeSection
                isReallySmallScreen={isReallySmallScreen}
                componentImageSrc="/assets/web-security-green.png"
                componentImageAlt=""
                title="Keep Your Documents Safe and Secure Using Decentralized Technology"
                description="Our innovative solution empowers individuals to manage their critical documents and control access for trusted organizations. PASS simplifies service access, enabling seamless documents requests and secure data sharing for a smoother support process."
                button="Request a Demo"
                href="mailto:hugh@codeforpdx.org"
                hasMargin
              />
              <HomeSection
                isReallySmallScreen={isReallySmallScreen}
                componentImageSrc="/assets/app-green.png"
                componentImageAlt=""
                title="An App for Caseworkers"
                description="PASS allows users to quickly and securely share documents of their clients within the app. The app helps caseworkers verify and share documents such as ID and proof of income while retaining total control of them."
                hasMargin
              />
              <HomeSection
                isReallySmallScreen={isReallySmallScreen}
                componentImageSrc="/assets/key-features-green.png"
                componentImageAlt=""
                title="Key Features"
              />
              <KeyFeatures
                isReallySmallScreen={isReallySmallScreen}
                icon={<SecurityIcon fontSize={iconSize} />}
                title="Secure Storage"
                description="Store vital documents like IDs, Social Security information, birth certificates, medical records, and bank statements in a valid digital format."
              />
              <KeyFeatures
                isReallySmallScreen={isReallySmallScreen}
                icon={<Diversity1Icon fontSize={iconSize} />}
                title="Nonprofit-Caseworker Integration"
                description="The platform facilitates smooth communication between nonprofit organizations, case workers, and the individuals they serve. It allows nonprofit organizations to maintain a contact list, and caseworkers are assigned contacts whose data they can access securely."
              />
              <KeyFeatures
                isReallySmallScreen={isReallySmallScreen}
                icon={<SupportIcon fontSize={iconSize} />}
                title="Support Service"
                description="Verified documents can be used to facilitate access to service such as housing support and shelter accommodation. The platform simplifies the process of submitting necessary documents for such services."
              />
            </div>
          </Stack>
        </section>
      </Box>
    </Container>
  );
};

export default Home;
