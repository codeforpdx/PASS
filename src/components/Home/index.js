import HomeSection from './HomeSection';
import KeyFeatures from './KeyFeatures';

/**
 * Components and functions related to Home within project PASS and used for
 * the Home Page
 *
 * @namespace Home
 */
export { HomeSection, KeyFeatures };
