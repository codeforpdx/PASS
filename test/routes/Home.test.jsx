import { render, cleanup } from '@testing-library/react';
import React from 'react';
import { expect, it, afterEach, describe } from 'vitest';
import { ThemeProvider } from '@mui/material/styles';

import Home from '../../src/routes/Home';
import theme from '../../src/theme';

const HomeWithContexts = () => (
  <ThemeProvider theme={theme}>
    <Home />
  </ThemeProvider>
);

describe('Home Page', () => {
  afterEach(() => {
    cleanup();
  });

  it('renders', () => {
    const { container } = render(<HomeWithContexts />);
    expect(container).toMatchSnapshot();
  });
});
