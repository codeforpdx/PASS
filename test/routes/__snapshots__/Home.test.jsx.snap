// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home Page > renders 1`] = `
<div>
  <main
    class="MuiContainer-root MuiContainer-maxWidthS css-1e1pk0w-MuiContainer-root"
  >
    <div
      class="MuiBox-root css-15kuacd"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation2 css-f5qjqr-MuiPaper-root"
      >
        <section
          id="home"
        >
          <h1>
            HOME PAGE
          </h1>
        </section>
      </div>
    </div>
  </main>
</div>
`;
